qcodes/dataset/experiment_container.py

```diff
@@ -326,28 +322,70 @@ def load_experiment_by_name(name: str,
 
 def load_or_create_experiment(experiment_name: str,
                               sample_name: Optional[str] = None,
-                              conn: Optional[ConnectionPlus]=None)->Experiment:
+                              load_last_duplicate: bool = False,
+                              conn: Optional[ConnectionPlus]=None,
+                              ) -> Experiment:
     """
     Find and return an experiment with the given name and sample name,
     or create one if not found.
 
     Args:
-        experiment_name: Name of the experiment to find or create
-        sample_name: Name of the sample
+        experiment_name: Name of the experiment to find or create.
+        sample_name: Name of the sample.
+        load_last_duplicate: To prevent raising error for having multiple
+            experiment with the same name and sample_name, and to load the
+            last duplicated experiment.
         conn: Connection to the database. If not supplied, a new connection
-          to the DB file specified in the config is made
+          to the DB file specified in the config is made.
 
     Returns:
         The found or created experiment
     """
     conn = conn or connect(get_DB_location())
     try:
         experiment = load_experiment_by_name(experiment_name, sample_name,
-                                             conn=conn)
+                                             load_last_duplicate=load_last_duplicate,
+                                             conn=conn,)
     except ValueError as exception:
         if "Experiment not found" in str(exception):
             experiment = new_experiment(experiment_name, sample_name,
                                         conn=conn)
         else:
             raise exception
     return experiment
+
+
+def _find_exp_rows(
```

Comment:
I think we should be able to move this function to the queries module where it really belongs

Reply:
hmm. you are right. 
do you have any suggestion for the name or think it is fine?

Reply:
I think we should also consider letting the function return a list of exp ids. That way we isolate this code further from sqlite

Reply:
We could call it `get_experiment_id_from_name_sample_name`  if that is not too verbose (assuming that we also change it to return a list of exp ids

Reply:
That would be also useful if it returns list of exp_ids, rather than the rows if we want to move it to queries and make it public. 
Then I need to change the way of deducting in line 307, which is easy.

Reply:
I think we don't need this function at all.
Looking into queries, I found this function `get_matching_exp_ids(conn, **match_conditions)`  which takes any kwarg. I tested it and it returns exp_id list for name and sample_name